### DISCUSSION TITLE
Package: tree-view-copy-relative-path

### DISCUSSION THREAD

**Discussion by Lonniebiz** [pulsar-edit/.github#233] [Community Package Help] [answered]
In Atom, there was this package called **tree-view-copy-relative-path** that was super useful in big projects having a deep tree structure.

This package caused an additional context-menu-item to be located in the context menu (that's produced when right-clicking a file from the tree on the left-side of atom); the menu option was "Copy Relative Path". Upon clicking it, it would copy that file's relative path (relative to the file you're currently editing in atom) into the clipboard. Then, you could paste into an include statement (for example). Super useful!

I love js, but those relative paths in include statements aren't fun:  `include { func1 } from '../../../../code.js'`, but this package made it easy to always paste in the right relative path. That's one package I definitely want in Pulsar too, but I don't see it listed.

I still have my .atom folder, and I'm sure this plugin must be buried somewhere within it. Can I transfer it over to the `.config/Plusar` directory somehow? If so, should I upload it somewhere so others can just download from within Plusar?

**Comment by savetheclocktower** [comment, marked as answer]
If you go into your Pulsar settings and switch to the Install pane, you should be able to search for `tree-view-copy-relative-path` and install it from there. I just tried it and it seems to install fine. (I see it listed [on the package site](https://web.pulsar-edit.dev/packages/tree-view-copy-relative-path), though you might get an error from that link — we're having some devops difficulties that are hard to troubleshoot.)

If that didn't work, you could install straight from GitHub:

```sh
ppm install vladimir-tikhonov/tree-view-copy-relative-path
```

And if _that_ didn't work… then, yes, you could copy from `~/.atom/packages/tree-view-copy-relative-path` to `~/.pulsar/packages/tree-view-copy-relative-path`. But if one of those other methods somehow doesn't work, please let us know.

**Reply by Lonniebiz** [reply]
It is indeed there. I was searching in the wrong place (under "Installed Packages", duh).

Hey, I noticed that pulsar has two locations for configs (it seems):
1) ~/.pulsar
2) ~/.config/Pulsar

Do happen to know why this is? I guess number 1 is the most important to backup?

**Reply by savetheclocktower** [reply]
I'm betting you're on Linux? There are some distros that encourage software to put all their configuration data in `~/.config`, but that's something that Electron hasn't really implemented yet. But, yes, `~/pulsar` is crucial to backup. [sync-settings](https://web.pulsar-edit.dev/packages/sync-settings) is a great way to do so.

**Reply by Daeraxa** [reply]
>     1. ~/.pulsar
> 
>     2. ~/.config/Pulsar
> 
> Do happen to know why this is? I guess number 1 is the most important to backup?

The former is your actual profile which is what you indeed would want to back up and this is more or less hard coded by Pulsar to go there (unless you use the ATOM_HOME env var. The latter is Pulsar's Electron cache and is transient (although it will store stuff like open tabs etc. in order to reload the state of the editor from when you closed it).

I'm keen to support the XDG spec in order to move everything into the prescribed locations. See https://github.com/orgs/pulsar-edit/discussions/68
